Documentation/Index.rst

```diff
@@ -0,0 +1,145 @@
+
+================
+Additional Notes
+================
+
+.. highlight:: shell
+.. default-role:: code
+
+
+Expose MySQL-like Database Server to the Host System
+====================================================
+
+You may want to use a non-default port like `4306` to avoid conflicts with the host system.
+Here is an example snippet from :file:`docker-compose.yml`::
+
+  db:
+    image: mariadb:10.1
+    environment:
+      - MYSQL_ROOT_PASSWORD=secret
+    volumes:
+      - ./mysql:/var/lib/mysql
+    ports:
+      - "4306:3306"
+
+
+Start daemonized and list containers::
+
+    cd docker-lemp
+    docker-compose up -d
+    docker ps
+
+Test the connection to the database server::
+
+    ➜  docker-lemp git:(master) ✗ mysql -h 127.0.0.1 -P 4306 -u root -psecret
+
+    Welcome to the MySQL monitor.  Commands end with ; or \g.
+    Your MySQL connection id is 3
+    Server version: 5.5.5-10.1.19-MariaDB-1~jessie mariadb.org binary distribution
+
+    Copyright (c) 2000, 2016, Oracle and/or its affiliates. All rights reserved.
+
+    Oracle is a registered trademark of Oracle Corporation and/or its
+    affiliates. Other names may be trademarks of their respective
+    owners.
+
+    Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.
+
+    mysql> quit
+    Bye
+
+Success!
+
+You can now use database connections. An example of Python-Flask notation is::
+
+    mysql+mysqldb://root:secret@127.0.0.1:4306/my_database_001
+
+
+.. attention::
+
+   Pitfall! **Do write `127.0.0.1`.** Do not write `localhost`!
+
+
+With `localhost` you'll get an ERROR similar to::
+
+    ➜  docker-lemp git:(master) ✗ mysql -h localhost -P 4306 -u root -psecret
+    ERROR 1045 (28000): Access denied for user 'root'@'localhost' (using password: YES)
+
+The reason for this is that with `localhost` a socket connection will be tried instead
+of a TCP/IP connection.
+
+
+Log all Database Queries
+========================
+
+While developing and debugging there may be times when you want to log all database queries.
+
+Make sure you can edit the configuration file :file:`/etc/mysql/my.cnf`::
+
+    # go to the shell of the db container
+    docker exec -it dockerlemp_db_1 bash
+
+    # update packages list
+    apt-get update
+
+    # install editor
+    apt-get install vim
+
+    # edit
+    vi /etc/mysql/my.cnf
+
+    # return
+    exit
+
+Read about `MariaDB's general logging facility <https://mariadb.com/kb/en/mariadb/general-query-log/>`__.
+In :file:`my.cnf` update the `mysqld` section::
+
+    [mysqld]
+    general-log
+    general-log-file=queries.log
+    log-output=file
+
+Restart the database container. Afterwards watch out for :file:`docker-lemp/mysql/queries.log` on the host system.
+
+
+phpMyAdmin
+==========
+
+Use an extra Docker container to run `phpMyAdmin`:
```

Comment:
It could be done in the same docker-compose project like you already showed in https://github.com/stucki/docker-lemp/pull/2/commits/f3d8e74bbc880c9b98c2619234a27c7b4f136881#diff-4e5e90c6228fd48698d074241c2ba760.

Here again, I think it's too detailed and not specifically about what I tried to do with this project.